"panic: runtime error: index out of range" adding a gateway to the default profile then modifying a node

### Steps to reproduce

On a clean build of a system with no nodes, adding the line
```
wwctl profile set -y default --netmask 255.255.0.0
```
the last command will error
```
wwctl node set -y --ipaddr $IP --hwaddr $MAC $NODE
wwctl node add $NODE
wwctl node set -y --container rocky-9 $NODE
```

Specifically, when the line in the following code is uncommented.
https://github.com/MiddelkoopT/ohpc-jetstream2/blob/0b8a5db0964f8ea325793e1e8ad8c8d62b922c3f/playbooks/ohpc-head.yaml#L160

To debug/get the error codes I ran the following playbooks from https://github.com/MiddelkoopT/ohpc-jetstream2/tree/0b8a5db0964f8ea325793e1e8ad8c8d62b922c3f/playbooks (see the repo for complete context) and ran the above commands manually with the $MAC, $IP, and $NODE values set.

```
ansible-playbook -v playbooks/ohpc-system.yaml
ansible-playbook -v playbooks/ohpc-head.yaml
```
Manually run commands from the following playbook:
https://github.com/MiddelkoopT/ohpc-jetstream2/blob/0b8a5db0964f8ea325793e1e8ad8c8d62b922c3f/playbooks/ohpc-nodes.yaml#L19





### Error message

```
            wwctl node set -y --ipaddr $IP --hwaddr $MAC $NODE
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/warewulf/warewulf/internal/pkg/node.(*NodeYaml).FindAllNodes(0xc00025f9b8)
	/usr/src/warewulf/internal/pkg/node/constructors.go:213 +0x1b65
github.com/warewulf/warewulf/internal/pkg/api/node.NodeSetParameterCheck(0xc00025fbc0, 0x0)
	/usr/src/warewulf/internal/pkg/api/node/node.go:229 +0x305
github.com/warewulf/warewulf/internal/pkg/api/node.NodeSet(0xc00025fbc0)
	/usr/src/warewulf/internal/pkg/api/node/node.go:194 +0x3f
github.com/warewulf/warewulf/internal/app/wwctl/node/set.CobraRunE.func1(0xc000120c00?, {0xc00007e6c0, 0x1, 0x4})
	/usr/src/warewulf/internal/app/wwctl/node/set/main.go:86 +0x8fc
github.com/spf13/cobra.(*Command).execute(0xc000120c00, {0xc00007e680, 0x4, 0x4})
	/usr/src/warewulf/vendor/github.com/spf13/cobra/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0x16326e0)
	/usr/src/warewulf/vendor/github.com/spf13/cobra/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(0xc0000061a0?)
	/usr/src/warewulf/vendor/github.com/spf13/cobra/command.go:968 +0x19
main.main()
	/usr/src/warewulf/cmd/wwctl/main.go:13 +0x25
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/warewulf/warewulf/internal/pkg/node.(*NodeYaml).FindAllNodes(0xc0002779b8)
	/usr/src/warewulf/internal/pkg/node/constructors.go:213 +0x1b65
github.com/warewulf/warewulf/internal/pkg/api/node.NodeSetParameterCheck(0xc000277bc0, 0x0)
	/usr/src/warewulf/internal/pkg/api/node/node.go:229 +0x305
github.com/warewulf/warewulf/internal/pkg/api/node.NodeSet(0xc000277bc0)
	/usr/src/warewulf/internal/pkg/api/node/node.go:194 +0x3f
github.com/warewulf/warewulf/internal/app/wwctl/node/set.CobraRunE.func1(0xc0000be900?, {0xc0000a1620, 0x1, 0x6})
	/usr/src/warewulf/internal/app/wwctl/node/set/main.go:86 +0x8fc
github.com/spf13/cobra.(*Command).execute(0xc0000be900, {0xc0000a15c0, 0x6, 0x6})
	/usr/src/warewulf/vendor/github.com/spf13/cobra/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0x16326e0)
	/usr/src/warewulf/vendor/github.com/spf13/cobra/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(0xc0000061a0?)
	/usr/src/warewulf/vendor/github.com/spf13/cobra/command.go:968 +0x19
main.main()
	/usr/src/warewulf/cmd/wwctl/main.go:13 +0x25
```


### Information on your system

ww4 is built from source, commit 88446f9520be6d5f70760fc240baf0136cd69498

with
```bash
        cd /usr/src
        git clone https://github.com/warewulf/warewulf.git
        cd warewulf
        make clean defaults \
          PREFIX=/usr \
          BINDIR=/usr/bin \
          SYSCONFDIR=/etc \
          DATADIR=/usr/share \
          LOCALSTATEDIR=/var/lib \
          SHAREDSTATEDIR=/var/lib \
          MANDIR=/usr/share/man \
          INFODIR=/usr/share/info \
          DOCDIR=/usr/share/doc \
          SRVDIR=/var/lib \
          TFTPDIR=/var/lib/tftpboot \
          SYSTEMDDIR=/usr/lib/systemd/system \
          BASHCOMPDIR=/etc/bash_completion.d/ \
          FIREWALLDDIR=/usr/lib/firewalld/services \
          WWCLIENTDIR=/warewulf
        make all
        make install
```
(see repo for more context)

```
[root@head warewulf]# wwctl version
wwctl version:	 4.5.0rc1-1.git_88446f95
rpc version: apiPrefix:"rc1" apiVersion:"1" warewulfVersion:"4.5.0rc1-1.git_88446f95"
```

```
[root@head warewulf]# cat /etc/os-release 
NAME="Rocky Linux"
VERSION="9.3 (Blue Onyx)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="9.3"
PLATFORM_ID="platform:el9"
PRETTY_NAME="Rocky Linux 9.3 (Blue Onyx)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:9::baseos"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2032-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-9"
ROCKY_SUPPORT_PRODUCT_VERSION="9.3"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.3"
```


### General information

- [X] I have run `wwctl version` and reported the contents of `/etc/os-release`
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have captured and reported relevant error messages and logs

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -70,6 +70,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Optionally detect network and netmask from CIDR IP address. #1016
 - Fix startup of wwclient on systemd hosts #1066
 - Change ownership of overlays to root:root #1086
+- Fix panic when node has no profile but network does #1094
 
 ### Changed
 

--- a/internal/app/wwctl/node/list/main_test.go
+++ b/internal/app/wwctl/node/list/main_test.go
@@ -125,6 +125,24 @@ nodes:
    - default
 `,
 		},
+		{
+			name:    "node list profile with network",
+			args:    []string{},
+			wantErr: false,
+			stdout: `  NODE NAME  PROFILES  NETWORK
+  n01        default         default
+`,
+			inDb: `WW_INTERNAL: 45
+nodeprofiles:
+  default:
+    network devices:
+      default:
+        device: eth0
+nodes:
+  n01:
+    profiles:
+    - default
+`},
 	}
 	conf_yml := `WW_INTERNAL: 0`
 	tempWarewulfConf, warewulfConfErr := os.CreateTemp("", "warewulf.conf-")

--- a/internal/pkg/node/constructors.go
+++ b/internal/pkg/node/constructors.go
@@ -205,7 +205,7 @@ func (config *NodeYaml) FindAllNodes() ([]NodeInfo, error) {
 		// set default/primary network is just one network exist
 		if len(n.NetDevs) >= 1 && !n.PrimaryNetDev.Defined() {
 			tmpNets := make([]string, 0, len(n.NetDevs))
-			for key := range node.NetDevs {
+			for key := range n.NetDevs {
 				tmpNets = append(tmpNets, key)
 			}
 			sort.Strings(tmpNets)

